GD32 build failure: CAN_MODE_NORMAL is redefined

**Describe the bug**
As first seen in #41673, GD32-based `longan_nano` board build fails because of redefinition of CAN_MODE_NORMAL.

```
In file included from /__w/zephyr/zephyr/tests/drivers/can/utilities/src/main.c:8:
/__w/zephyr/zephyr/include/zephyr/drivers/can.h:89: error: "CAN_MODE_NORMAL" redefined [-Werror]
   89 | #define CAN_MODE_NORMAL     0
      | 
In file included from /__w/zephyr/modules/hal/gigadevice/gd32vf103/standard_peripheral/include/gd32vf103_libopt.h:41,
                 from /__w/zephyr/modules/hal/gigadevice/gd32vf103/riscv/include/gd32vf103.h:237,
                 from /__w/zephyr/zephyr/soc/riscv/riscv-privilege/gd32vf103/soc.h:23,
                 from /__w/zephyr/zephyr/include/zephyr/arch/riscv/arch.h:28,
                 from /__w/zephyr/zephyr/include/zephyr/arch/cpu.h:25,
                 from /__w/zephyr/zephyr/include/zephyr/kernel_includes.h:33,
                 from /__w/zephyr/zephyr/include/zephyr/kernel.h:17,
                 from /__w/zephyr/zephyr/include/zephyr/init.h:11,
                 from /__w/zephyr/zephyr/include/zephyr/device.h:29,
                 from /__w/zephyr/zephyr/include/zephyr/drivers/can.h:12,
                 from /__w/zephyr/zephyr/tests/drivers/can/utilities/src/main.c:8:
/__w/zephyr/modules/hal/gigadevice/gd32vf103/standard_peripheral/include/gd32vf103_can.h:657: note: this is the location of the previous definition
  657 | #define CAN_MODE_NORMAL                    ((uint8_t)0x02U)             /*!< CAN normal mode */
```
Relevant PR that introduced the redefinition: #45380

**To Reproduce**
```
scripts/twister -T tests/drivers/build_all/modem -p longan_nano --build-only
```

--- a/soc/arm/gigadevice/gd32f403/soc.h
+++ b/soc/arm/gigadevice/gd32f403/soc.h
@@ -14,6 +14,10 @@
 
 #ifndef _ASMLANGUAGE
 #include <gd32f403.h>
+
+/* The GigaDevice HAL headers define this, but it conflicts with the Zephyr can.h */
+#undef CAN_MODE_NORMAL
+
 #endif /* !_ASMLANGUAGE */
 
 #endif /* _SOC__H_ */

--- a/soc/arm/gigadevice/gd32f4xx/soc.h
+++ b/soc/arm/gigadevice/gd32f4xx/soc.h
@@ -10,6 +10,9 @@
 
 #include <gd32f4xx.h>
 
+/* The GigaDevice HAL headers define this, but it conflicts with the Zephyr can.h */
+#undef CAN_MODE_NORMAL
+
 #endif /* _ASMLANGUAGE */
 
 #endif /* _SOC_ARM_GIGADEVICE_GD32F4XX_SOC_H_ */

--- a/soc/riscv/riscv-privilege/gd32vf103/soc.h
+++ b/soc/riscv/riscv-privilege/gd32vf103/soc.h
@@ -21,6 +21,10 @@
 #ifndef _ASMLANGUAGE
 #include <zephyr/toolchain.h>
 #include <gd32vf103.h>
+
+/* The GigaDevice HAL headers define this, but it conflicts with the Zephyr can.h */
+#undef CAN_MODE_NORMAL
+
 #endif  /* !_ASMLANGUAGE */
 
 #endif  /* RISCV_GD32VF103_SOC_H */

